src/dbnode/storage/namespace/schema.go

```diff
@@ -41,25 +41,24 @@ var (
 )
 
 type schemaDescr struct {
-	deployId string
-	md       *desc.MessageDescriptor
+	deployId     string
+	prevDeployId string
+	md           *desc.MessageDescriptor
 }
 
 func (s *schemaDescr) DeployId() string {
 	return s.deployId
 }
 
+func (s *schemaDescr) PrevDeployId() string {
+	return s.prevDeployId
+}
+
 func (s *schemaDescr) Equal(o SchemaDescr) bool {
-	if s == nil && o == nil {
-		return true
-	}
-	if s != nil && o == nil || s == nil && o != nil {
-		return false
-	}
 	if _, ok := o.(*schemaDescr); !ok {
```

Comment:
Do we still need this check?

Reply:
yes

Reply:
I see, you mean that we only invoke interface method for equality check. yes, maybe we don't need the assertion. 

in that case, I will have to add back the nil check.